addons/website_event/static/src/js/website_event.js

```diff
@@ -43,23 +42,25 @@ var EventRegistrationForm = publicWidget.Widget.extend({
     on_click: function (ev) {
         ev.preventDefault();
         ev.stopPropagation();
-        var $form = $(ev.currentTarget).closest('form');
-        var $button = $(ev.currentTarget).closest('[type="submit"]');
+        const formEl = ev.currentTarget.closest("form");
+        const buttonEl = ev.currentTarget.closest("[type='submit']");
         const post = this._getPost();
-        $button.attr('disabled', true);
-        return rpc($form.attr('action'), post).then(function (modal) {
-            var $modal = $(modal);
-            $modal.find('.modal-body > div').removeClass('container'); // retrocompatibility - REMOVE ME in master / saas-19
-            $modal.appendTo(document.body);
-            const modalBS = new Modal($modal[0], {backdrop: 'static', keyboard: false});
-            modalBS.show();
-            $modal.appendTo('body').modal('show');
-            $modal.on('click', '.js_goto_event', function () {
-                $modal.modal('hide');
-                $button.prop('disabled', false);
+        buttonEl.disabled = true;
+        return rpc(formEl.action, post).then(function (modal) {
+            const modalEl = new DOMParser().parseFromString(modal, "text/html").body.firstChild;
+            modalEl.querySelector(".modal-body > div").classList.remove("container"); // retrocompatibility - REMOVE ME in master / saas-19
+            document.body.append(modalEl);
+            const formModal = Modal.getOrCreateInstance(modalEl, {
+                backdrop: "static",
+                keyboard: false,
             });
-            $modal.on('click', '.btn-close', function () {
-                $button.prop('disabled', false);
+            formModal.show();
+            modalEl.querySelector(".js_goto_event").addEventListener("click", function () {
+                formModal.hide();
+                buttonEl.disabled = false;
+            });
+            modalEl.querySelector(".btn-close").addEventListener("click", function () {
+                buttonEl.disabled = false;
```

Comment:
Self reply: no they shouldn't, it's hardly possible to keep track of all possible `modalEl`s created if you cancel your registration once or more. But it could potentially be destroyed inside the event listener itself.